File: application/backend-credit-card-enrollment/backend-c#/Common/SessionAuth/SessionService.cs
namespace CreditCardEnrollment.Common.SessionAuth;

public class SessionService {
    private readonly SessionRepository _sessionRepository;
    private readonly int _sessionExpirationSeconds;

    public SessionService(SessionRepository sessionRepository, int sessionExpirationSeconds) {
        _sessionRepository = sessionRepository;
        _sessionExpirationSeconds = sessionExpirationSeconds;
    }

    public string AuthenticatedUserIdFromSessionToken(string sessionToken) {
        return _sessionRepository.AuthenticatedUserIdFromSessionToken(sessionToken, _sessionExpirationSeconds) 
               ?? throw new Exception("Invalid session");
    }
}

File: application/backend-credit-card-enrollment/backend-typescript/src/common/sessionAuth/SessionService.ts
import { SessionRepository } from './SessionRepository';

export class SessionService {
    private readonly sessionRepository: SessionRepository;
    private readonly sessionExpirationSeconds: number;

    constructor(sessionRepository: SessionRepository, sessionExpirationSeconds: number) {
        this.sessionRepository = sessionRepository;
        this.sessionExpirationSeconds = sessionExpirationSeconds;
    }

    async authenticatedUserIdFromSessionToken(sessionToken: string): Promise<string> {
        const userId = await this.sessionRepository.authenticatedUserIdFromSessionToken(
            sessionToken,
            this.sessionExpirationSeconds
        );

        if (!userId) {
            throw new Error('Invalid session');
        }

        return userId;
    }
}
